### ISSUE TITLE
Pre-trained Model

### ISSUE THREAD

**Issue by ant1pink** [Abhijit-2592/Keras_object_detection#1]
Hello there,

I had a bad experience on yhenon's repo. Would you please share a pre-trained model based on the dataset voc2007 or 2012 to let me have a try?

Many thanks,

Ray

**Comment by Abhijit-2592** [comment]
Actually I don't have any pre trained models on public datasets. For my specific case, What I did was, I converted a classification problem to a detection problem. In the sense first I trained a the network as a classification network and then initialised the faster rcnn with those weights and trained. This is how the object detection networks are trained in general. Try this out and lemme know .

Update: if for some reason you cannot train a classification network on your dataset initialize the network using imagenets weights. Don't train with fully random initialization

**Comment by ant1pink** [comment]
Thanks very much for these details. The pretrained weights I am using is resnet50 (https://github.com/fchollet/deep-learning-models/releases/download/v0.2/resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5)
But it doesn't work very well.

**Comment by Abhijit-2592** [comment]
Kindly share your loss plots?

**Comment by ant1pink** [comment]
Sorry, I am new to this area. And thank you for your patience.
I haven't plotted it out, but i got this 
<img width="279" alt="wechat image_20180110171347" src="https://user-images.githubusercontent.com/28095937/34785628-a2493814-f629-11e7-8617-8b7ece61a206.png">

The problem I got is there is no bounding box on test image.

I guess the problem is that I used num_epochs =3 and epoch_length = 20. Probably, it is not enough to train a model to predict bounding box on test image.

**Comment by ant1pink** [comment]
And the dataset I am using is Flickr logos 27 (http://image.ntua.gr/iva/datasets/flickr_logos/)
In this case, should I use the pretrained weights or should I  train a classification network on this dataset  to initialize the network?
Thanks,
Ray

**Comment by Abhijit-2592** [comment]
First of all You need to train for more epochs. Look at your RPN loss. It is not able to propose regions at all. Train them for more epochs. From seeing the elapsed time it is clear that you are training on a CPU. Use a GPU or train in cloud. Training FRCNN in CPUs is not advisable.

Looking at your dataset, initializing the model with imagenet weights should give a good edge

**Comment by ant1pink** [comment]
Hi, I followed your instruction and finally got some results. 
![image](https://user-images.githubusercontent.com/28095937/34907834-95f0b254-f87c-11e7-8e19-6ed24b20b43e.png)
But however it is not every accurate, most of logos can not be detected at the moment. It is probably because:
1. There are only 700 training images for 20 classes. 
2. The epochs I used for training are still too less. I used 30 epochs, each epoch length is 1000.

I want to increase the accuracy, do you have any suggestions?

Thanks,
R

**Comment by Abhijit-2592** [comment]
It's hard to say without looking into the dataset. I haven't had time to look at the dataset :sweat_smile:

**Comment by Abhijit-2592** [comment]
1. May be try some brute force methods like train for high number of epochs like 100 + and check at which point the network starts over-fitting. 
2. Try different network architectures like inception-v3,Resnet-etc
3. Try 4 step alternative training
4. different inititalizations

It is hard to say which will work. Spend more time with your data. Try understanding it. Use tools like t-sne to visualize high dimensional data points

Happy training :)

**Comment by ant1pink** [comment]
Thanks a lot, I will let you know if I get some progresses 👍 

Thanks,
R

**Comment by ant1pink** [comment]
![image](https://user-images.githubusercontent.com/28095937/34939229-fe136ec8-f9e2-11e7-9581-853318230ac6.png)
Wow, it works much better now. Thanks very much!!!!

**Comment by deepaklunked** [comment]
@ant1pink 
What spefific changes did you do after Abhijit's comment? Like how many epochs, which network arch, etc..

**Comment by ant1pink** [comment]
@deepaklunked I am using VGG now, it seems that it doesn't have to be too many epochs. But as Abhijit-2592 mentioned, it totally depends your data. For my case, after 30 epochs, my RPN loss doesnt decrease significantly anymore.

**Comment by Abhijit-2592** [comment]
@ant1pink  Glad it worked. Looks like you have solved your problem. I am closing this issue now. Please feel free to open any-other issues in future if any.
